README.md

```diff
@@ -2,34 +2,34 @@
 
 [![Build Status](https://travis-ci.org/tapjs/foreground-child.svg)](https://travis-ci.org/tapjs/foreground-child) [![Build status](https://ci.appveyor.com/api/projects/status/kq9ylvx9fyr9khx0?svg=true)](https://ci.appveyor.com/project/isaacs/foreground-child)
 
-Run a child as if it's the foreground process.  Give it stdio.  Exit
-when it exits.
+Run a child as if it's the foreground process. Give it stdio.
 
 Mostly this module is here to support some use cases around wrapping
 child processes for test coverage and such.
 
 ## USAGE
 
 ```js
-var foreground = require('foreground-child')
+const fg = require('foreground-child')
 
 // cats out this file
-var child = foreground('cat', [__filename])
+const {child, close} = fg('cat', [__filename])
 
 // At this point, it's best to just do nothing else.
 // return or whatever.
 // If the child gets a signal, or just exits, then this
 // parent process will exit in the same way.
-```
 
-A callback can optionally be provided, if you want to perform an action
-before your foreground-child exits:
+// Close is a promise resolved when the child is closed.
+// You can use it to inspected its exit code and exit the parent process.
+close.then((close) => {
+  // You can use this handler to perform an action before exiting the
+  // foreground child.
+  console.log(close.code);
```

Comment:
How's `close` a function and object at same time?

Reply:
Well, it's like most CJS exports: a function with additional properties (`fg()` and `fg.spawn()`, `signalExit()` and `signalExit.signals()`, etc.).

It boils down to:
```javascript
const close = () => { ... };
close.code = code;
close.signal = signal;
return close;
```

Reply:
Oh, got it. Thanks!